maps/biter_battles_v2/mirror_terrain.lua

```diff
@@ -117,13 +117,12 @@ function Public.remove_hidden_tiles(event)
 		area = bb,
 		has_hidden_tile = true,
 		name = "stone-path",
+		name = "refined-concrete"
 	}
 
-	local tiles = {}
 	for i, tile in pairs(to_remove) do
 		if not tile.valid then goto remove_hidden_cont end
-		surface.set_hidden_tile(tile.position, nil)
-
+		surface.set_hidden_tile(tile.position, surface.get_hidden_tile(surface.get_tile(tile.position.x,-tile.position.y).position))
```

Comment:
```suggestion
		surface.set_hidden_tile(tile.position, surface.get_hidden_tile(surface.get_tile(tile.position.x,-tile.position.y - 1).position))
```
when mirroing a tile, you need to subtract 1 (see Public.invert_tiles) because the coordinate of a tile is its upper left corner, and if you don't subtract minus 1, then the mirrored coordinate becomes the lower left corner, which is the coordinate of another tile. I think this is the reason for the inability to remove the tile near river.

Reply:
also this one line is too long

Reply:
Too long? Like editor-wise?

Reply:
yes, there is a lot going on in it

Reply:
Okay, after your suggestion, indeed I don't have this issue anymore of tile along river NOK.
Also for the one liner too long, I splitted it in 2, is it better now?

Reply:
Better than it was. Don't you think that this was too complicated code for one line

Reply:
Yes indeed, too many things in just 1 line (function calling function calling function)

Reply:
Think issue solved now, resolving it.